openedx/core/lib/block_structure/transformer_registry.py

```diff
@@ -31,6 +35,22 @@ def get_registered_transformers(cls):
             return set()
 
     @classmethod
+    @memoized
+    def get_write_version_hash(cls):
+        """
+        Returns a deterministic hash value of the WRITE_VERSION of all
+        registered transformers.
+        """
+        hash_obj = sha1()
+
+        sorted_transformers = sorted(cls.get_registered_transformers(), key=lambda t: t.name())
+        for transformer in sorted_transformers:
+            hash_obj.update(transformer.name())
+            hash_obj.update(unicode(transformer.WRITE_VERSION))
```

Comment:
hashlib operates on bytes, not unicode objects.  These should be explicitly encoded so as not to break compatibility when 1. passed non-ascii strings or 2. edx-platform (or block_structure) is converted to Python 3.

An example of broken behavior in Python 3:
```
Python 3.5.0 (default, Sep 23 2015, 04:42:00) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.72)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from hashlib import sha1
>>> sha1('abc')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Unicode-objects must be encoded before hashing
>>> sha1('abc'.encode('ascii')
<sha1 HASH object @ 0x106e3c990>
```

Or in Python 2:

```
Python 2.7.12 (default, Oct 11 2016, 05:24:00) 
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.38)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from hashlib import sha1
>>> sha1(u'abc')  # this is okay
<sha1 HASH object @ 0x106e3c990>
>>> sha1(u'abc\N{SNOWMAN}')  # but this fails
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2603' in position 3: ordinal not in range(128)
>>> sha1(u'abc\N{SNOWMAN}'.encode('utf-8'))  # this works too
<sha1 HASH object @ 0x106e3c9e0>
```

Option 1 is unlikely, because transformer names are tags that will almost certainly always be ascii-encodable, and WRITE_VERSION is an integer, but Option 2 is a relevant eventuality.